src/app/controllers/encrypt-submissions.server.controller.js

```diff
@@ -227,11 +227,33 @@ exports.getMetadata = function (req, res) {
   let { page, submissionId } = req.query || {}
   let numToSkip = parseInt(page - 1 || 0) * pageSize
 
+  if (page == -1) {
+    numToSkip = 0
+    page = 0
+    pageSize = 1000000000 // effectively unlimited
+  }
+
+  if (
+    isMalformedDate(req.query.startDate) ||
+    isMalformedDate(req.query.endDate)
+  ) {
+    return res.status(StatusCodes.BAD_REQUEST).json({
+      message: 'Malformed date parameter',
+    })
+  }
+
   let matchClause = {
     form: req.form._id,
     submissionType: 'encryptSubmission',
   }
 
+  const augmentedQuery = createQueryWithDateParam(
+    req.query.startDate,
+    req.query.endDate,
+  )
+
+  matchClause = { ...matchClause, ...augmentedQuery }
```

Comment:
code is cleaner if we rearrange to define augmentedQuery before matchClause

Reply:
Done.